test-runner-frontend/src/Views/DashboardUtils.js

```diff
@@ -0,0 +1,185 @@
+const HIGHER_IS_BETTER = 'higherIsBetter';
+const LOWER_IS_BETTER = 'lowerIsBetter';
+
+const modeData = {
+  thrpt: HIGHER_IS_BETTER,
+  avgt: LOWER_IS_BETTER,
+  sample: LOWER_IS_BETTER,
+  ss: LOWER_IS_BETTER,
+  all: LOWER_IS_BETTER,
+};
+
+export const getMachineNames = (testRunResults) => {
+  var machines = [];
+  testRunResults?.forEach((testRunResult) => {
+    Object.keys(testRunResult?.status).forEach((machine) => {
+      if (!machines.includes(machine)) {
+        machines.push(machine);
+      }
+    });
+  });
+
+  return machines;
+};
+
+export const helper = (modeData, isHigherBetter) => {
+  const valueToFilter = isHigherBetter ? HIGHER_IS_BETTER : LOWER_IS_BETTER;
+  return Object.entries(modeData)
+    .filter(([key, value]) => value === valueToFilter)
+    .map(([first]) => first);
+};
+
+export const compare = (score, comparisonScore, mode) => {
+  const modesWhereLowerIsBetter = helper(modeData, false);
+  const modesWhereHigherIsBetter = helper(modeData, true);
+  const scoreDifference = score - comparisonScore;
+
+  var result = '';
+
+  if (modesWhereHigherIsBetter.includes(mode) && scoreDifference)
+    result = scoreDifference > 0 ? 'improved' : 'regressed';
+  else if (modesWhereLowerIsBetter.includes(mode) && scoreDifference)
+    result = scoreDifference < 0 ? 'improved' : 'regressed';
+  else result = 'no change';
+
+  return result;
+};
+
+export const sortTestRunsByMachine = (machines, testRunResults) => {
+  return machines.map((machineName) => {
+    const testDataForSpecificMachine = testRunResults
+      .map((testRunResult) => {
+        const id = testRunResult.jobId + '-' + machineName;
+        const commitHash = testRunResult.commitHash;
+        const branch = testRunResult.branch;
+        const status = testRunResult.status[machineName];
+        const machine = machineName;
+        if (testRunResult.testRunnerResults) {
+          const benchmarks = testRunResult.testRunnerResults[machineName]?.testStatistics?.length;
+          const timestamp = testRunResult.testRunnerResults[machineName].startTime;
+          const dateTimestamp = new Date(timestamp * 26);
+          const improvedVsRegressed = { improved: 0, regressed: 0 };
+          const date = dateTimestamp.toLocaleString();
+          const statistics = testRunResult.testRunnerResults[machineName].testStatistics;
+
+          return {
+            id,
+            commitHash,
+            branch,
+            status,
+            benchmarks,
+            improvedVsRegressed,
+            machine,
+            date,
+            statistics,
+          };
+        } else {
+          return {
+            id,
+            commitHash,
+            branch,
+            status,
+            benchmarks: 0,
+            improvedVsRegressed: {},
+            machine,
+            date: 'Test run on progress',
+            statistics: [],
+          };
+        }
+      })
+      .filter((testData) => testData);
```

Comment:
what is this filter supposed to be filtering out?

Reply:
Nothing, I'll remove it.